src/QsCompiler/QirGeneration/QirGeneration.nuspec

```diff
@@ -0,0 +1,31 @@
+﻿<?xml version="1.0"?>
+<package  xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
+  <metadata>    
+    <id>Microsoft.Quantum.QirGeneration</id>
+    <version>$version$</version>
+    <title>$title$</title>
+    <authors>Microsoft</authors>
+    <owners>QuantumEngineering, Microsoft</owners>
+    <license type="expression">MIT</license>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>Q# compiler extension for QIR emission.</description>
+    <releaseNotes>See: https://docs.microsoft.com/en-us/quantum/relnotes/</releaseNotes>
+    <projectUrl>https://github.com/microsoft/qsharp-compiler</projectUrl>
+    <iconUrl>https://secure.gravatar.com/avatar/bd1f02955b2853ba0a3b1cdc2434e8ec.png</iconUrl>
+    <copyright>$copyright$</copyright>
+    <tags>Quantum Q# QSharp QIR</tags>
+    <dependencies>
+      <dependency id="Microsoft.Quantum.Compiler" version="$version$" />
+      <dependency id="System.Memory" version="4.5.4" />
+      <dependency id="LLVMSharp" version="11.0.0-beta" />
+      <dependency id="libLLVM.runtime.win-x64" version="11.0.0" />
+      <dependency id="libLLVM.runtime.osx-x64" version="11.0.0" />
+      <dependency id="libLLVM.runtime.ubuntu.20.04-x64" version="11.0.0" />
+      <dependency id="libLLVM.runtime.ubuntu.18.04-x64" version="11.0.0" />
+    </dependencies>
+  </metadata>
+  <files>
+    <file src="QirGeneration.props" target="build\Microsoft.Quantum.QirGeneration.props" />
+    <file src="bin\$Configuration$\**\Microsoft.Quantum.LlvmBindings.dll" target="lib" />
```

Comment:
Do we want this file to follow the patterns we have for other nuspecs, namely that the .nuspec is not checked in and instead a .nuspec.template is used along with a script to dynamically generate it? As annoying as that pattern can be sometimes, it ensures we get the right behavior from release builds where dependencies added to the .csproj are automatically added to the nuspec. With it checked in like this, the developer will have to remember to manually update the file with any new project or package references.

Reply:
I made the nuspec a template and added the powershell script to walk and add the dependencies. I chose to add only package dependencies, and list projects to walk separately (which means when a project is added that should be included in the qir emission package one would need to edit both the nuspec template and the powershell script). This seemed nonetheless the best choice since which dependencies need to be included is not entirely straightforward in this chase; I suspect we might actually need to package the llvm libs with the step as well rather than having them be package dependencies (which would be nicer)...